src/app/components/home-carousel/index.js

```diff
@@ -0,0 +1,142 @@
+import React, { Component } from 'react';
+import classnames from 'classnames';
+import transitionOnScroll from 'app/lib/transition-on-scroll';
+
+import ScrollWrapper from 'app/components/scroll-wrapper';
+import TimerUI from 'app/components/timer-ui';
+import Video from 'app/components/video';
+
+const itemsRefreshInterval = 8000;
+const tickerFrequency = 50;
+const transitionDuration = 300;
+const distance = 60;
+
+function goToNextItems(component) {
+  component.setState({ tick: itemsRefreshInterval });
+
+  if (component.state.currentStartItem === component.props.carouselItems.length - 2) {
+    component.setState({ currentStartItem: 0 });
+  } else {
+    component.setState({ currentStartItem: component.state.currentStartItem + 2 });
+  }
+}
+
+class HomeCarousel extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      currentStartItem: 0,
+      tick: itemsRefreshInterval,
+      paused: false
+    }
+  }
+
+  ticker() {
+    /* Start the ticker when half of the carousel comes into view and stop it when it's off out */
+    if (this.props.scrollProgress >= 0.25 && this.props.scrollProgress <= 0.75) {
+
+      if (this.state.tick === 0) {
+        goToNextItems(this);
+      }
+
+      if (!this.state.paused) {
+        this.setState({ tick: this.state.tick - tickerFrequency });
```

Comment:
State changes can be batched for performance reasons, so it's better to use a function to calculate the next state.

>Because this.props and this.state may be updated asynchronously, you should not rely on their values for calculating the next state.

See https://facebook.github.io/react/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous